modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -17,61 +17,125 @@
 namespace cv {
 namespace gapi {
 
+/**
+* \addtogroup gapi_serialization
+* @{
+*/
+
 namespace detail {
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS cv::GComputation getGraph(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS cv::GMetaArgs getMetaArgs(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS cv::GRunArgs getRunArgs(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS std::vector<std::string> getVectorOfStrings(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     template<typename... Types>
     cv::GCompileArgs getCompileArgs(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     template<typename RMatAdapterType>
     cv::GRunArgs getRunArgsWithRMats(const std::vector<char> &p);
 } // namespace detail
 
+/** @brief This function allows to serialize GComputation.
+ *
+ * Check different overloads for more examples.
+ * @param c GComputation to serialize.
+ * @return serialized vector of bytes.
+ */
 GAPI_EXPORTS std::vector<char> serialize(const cv::GComputation &c);
-//namespace{
 
-template<typename T> static inline
-T deserialize(const std::vector<char> &p);
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const cv::GCompileArgs& ca);
 
-//} //ananymous namespace
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const cv::GMetaArgs& ma);
 
-GAPI_EXPORTS std::vector<char> serialize(const cv::GCompileArgs&);
-GAPI_EXPORTS std::vector<char> serialize(const cv::GMetaArgs&);
-GAPI_EXPORTS std::vector<char> serialize(const cv::GRunArgs&);
-GAPI_EXPORTS std::vector<char> serialize(const std::vector<std::string>&);
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const cv::GRunArgs& ra);
 
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const std::vector<std::string>& vs);
+
+/**
+ * @private
+ */
+template<typename T> static inline
+T deserialize(const std::vector<char> &p);
+
+/** @brief This function allows to deserialize GComputation.
+ *
+ * Check different overloads for more examples.
+ * @param p serialized vector of bytes.
+ * @return deserialized GComputation object.
+ */
 template<> inline
 cv::GComputation deserialize(const std::vector<char> &p) {
```

Comment:
@alalek could you help a little? Do you know how to correctly document template specialization (or whatever the issue causes docs corruption)? Currently it looks like this: https://pullrequest.opencv.org/buildbot/export/pr/20163/docs/db/d06/group__gapi__serialization.html

Reply:
I believe it looks ok in your example - isn't it?

Reply:
(I see only two derializes there though)

Reply:
Let's ignore this for now. I will create a separate task to fix it later